src/modules/mc_pos_control/PositionControl/PositionControl.cpp

```diff
@@ -317,12 +317,11 @@ void PositionControl::_velocityControl(const float dt)
 	thrust_max_NE = math::min(thrust_max_NE_tilt, thrust_max_NE);
 
 	// Saturate thrust in NE-direction.
-	Vector2f thrust_sp_xy(_thr_sp);
+	const Vector2f thrust_sp_xy(_thr_sp);
+	const float thrust_sp_xy_norm = thrust_sp_xy.norm();
 
-	if (thrust_sp_xy.norm_squared() > thrust_max_NE * thrust_max_NE) {
-		float mag = thrust_sp_xy.length();
-		_thr_sp(0) = _thr_sp(0) / mag * thrust_max_NE;
-		_thr_sp(1) = _thr_sp(1) / mag * thrust_max_NE;
+	if (thrust_sp_xy_norm > thrust_max_NE) {
```

Comment:
use 
```
    bool longerThan(Type testVal) const {
        return norm_squared() > testVal*testVal;
    }
```
from the matrix library

Reply:
I thought about that but the norm anyways needs to be calculated... IMO it doesn't help readability and is less efficient but if it helps in your eyes we can put:
```
const Vector2f thrust_sp_xy(_thr_sp);

if (_thr_sp.xy().longerThan(thrust_max_NE)) {
	_thr_sp.xy() = Vector2f(_thr_sp).normalized() * thrust_max_NE;
}
```

Reply:
Like we just discussed it doesn't help in that particular case.